### PR TITLE
Add API call for Arm64 Sve.LoadVectorNonFaulting

### PR DESCRIPTION
**Pull request by a74nh**
Adds everything from the API down to calling the codegen.

LoadVectorNonFaulting() was chosen as it has been approved and requires no "hidden" mask nodes.

### CODE REVIEW

**Comment by a74nh** [comment]
> When run on real SVE hardware, the test fails because the jit allocates `Z16` as the mask (due to having no predicate register allocation). The emit functions treat this as `P16`, and then assert fail because the max predicate register is `P15`. Ideally, we should add some code to limit the mask to `Z15`, but I'm not sure where to do that.

I think I can see where this is done in lsra. Will add something.....

**Comment by kunalspathak** [comment]
> The Sve API needs marking experimental (I couldn't find the exact tag).

`[System.Runtime.Versioning.RequiresPreviewFeaturesAttribute("Sve is in preview.")]`

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    /// <summary>
+    /// This class provides access to the ARM SVE hardware instructions via intrinsics
+    /// </summary>
+    [Intrinsic]
+    [CLSCompliant(false)]
+    public abstract class Sve : AdvSimd
+    {
+        internal Sve() { }
+
+        public static new bool IsSupported { get => IsSupported; }
```
Comment (kunalspathak) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:20]: we will have to make sure to return `false` for Mono

File: src/coreclr/jit/emitarm64.cpp [hunk 2]
```diff
@@ -9924,6 +9952,12 @@ void emitter::emitIns_R_R_R(instruction     ins,
             fmt = IF_SVE_CZ_4A;
             break;
 
+        case INS_sve_ldnf1b:
+        case INS_sve_ldnf1h:
+        case INS_sve_ldnf1w:
+        case INS_sve_ldnf1d:
+            return emitIns_R_R_R_I(ins, size, reg1, reg2, reg3, 0, opt);
```
Comment (kunalspathak) [inline src/coreclr/jit/emitarm64.cpp:9959]: this doesn't look right. The caller should make sure to call appropriate `emitIns*` method.

**Review by kunalspathak — COMMENTED** [review]
Once the API is approved, we will have a top level issue with check boxes for the APIs, similar to how we have for https://github.com/dotnet/runtime/issues/93095 to track the progress. There, we will upload the autogenerated boilerplate code like:
- hwintrinsiclistarm64sve.h
- Sve.cs
- Sve.PlatformNotSupported.cs
- System.Runtime.Intrinsic.cs
- test templates

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs (hunk 1 above)
Comment (kunalspathak) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:21]: @fanyang-mono - do you know?

Comment (kunalspathak) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:21]: > One way of doing it is to add a new element to this array

aren't these the entries of things that are supported? so probably no SVE entry is needed in that array?

Comment (fanyang-mono) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:21]: When you specify `unsupported`, `SN_get_IsSupported` will return false. So it is needed.

Comment (a74nh) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:21]: Thanks! I can also see some examples of unsupported in `supported_x86_intrinsics`.

**Comment by a74nh** [comment]
With the mono changes added I built mono and ran the testsuite according to mono/testing.md

Looking in the test logs:
```
13:13:26.469 Running test: JIT/HardwareIntrinsics/Arm/Sve/Sve_r/Sve_r.dll
Supported ISAs:
  AdvSimd:   False
  Aes:       False
  ArmBase:   True
  Crc32:     False
  Dp:        False
  Rdm:       False
  Sha1:      False
  Sha256:    False
  Sve:       False
```

**Comment by a74nh** [comment]
This has been replaced with https://github.com/dotnet/runtime/pull/98218

**Pull request closed without merging**